src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -142,6 +143,24 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         });
     }
 
+    void TermControlAutomationPeer::NotifyNewOutput(const std::wstring_view newOutput)
+    {
+        auto dispatcher{ Dispatcher() };
+        if (!dispatcher)
+        {
+            return;
+        }
+        dispatcher.RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [weakThis{ get_weak() }, notificationText{ hstring(newOutput) }]() {
+            if (auto strongThis{ weakThis.get() })
+            {
+                strongThis->RaiseNotificationEvent(AutomationNotificationKind::ActionCompleted,
+                                                   AutomationNotificationProcessing::ImportantAll,
+                                                   notificationText,
+                                                   L"TerminalTextOutput");
+            }
+        });

```

Comment:
Don't you need to react to back-pressure here? What if you're called really fast on a slow system? Won't the background jobs just pile up more and more?

Reply:
We need the jobs to pile up. The screen reader wants to receive the entirety of the output.

Converting this thread to track synchronous notification dispatching. There's a bit of a concern here that the notification events being dispatched in an async way may be processed out of order.

Reply:
> We need the jobs to pile up.

This could be a matter of semantics, but to be specific, I think "piling them up" is not a good idea and needs to be prevented in this PR.

Windows Terminal can write the famous `big.txt` within half a second, but it has 128457 lines of text. Do we really want 128457 asynchronous jobs to be put into the job queue? This is why I mentioned earlier we need to control for back pressure. Personally I can imagine two solutions:
* A semaphore to let it run ahead by let's say one viewport height
* Only write the next line to the screen when the previous one finished being narrated (a semaphore of size 1)

I can imagine that a user of a narrator might want to have it finish the execution as fast as possible, but I'm worried that creating 128457 isn't a viable solution either.

Reply:
Would it be acceptable to make it read out something like "128,000 lines skipped"?

Reply:
How would an out-of-band announcement like that distinguish itself from output text?

Reply:
Good question... I'm just throwing ideas at the wall and seeing if anything sticks. Reading 128,000 lines seems intense.